go/libraries/doltcore/sqle/index/index_lookup.go

```diff
@@ -244,6 +245,19 @@ func (il *doltIndexLookup) Ranges() sql.RangeCollection {
 // Between returns whether the given types.Value is between the bounds. In addition, this returns if the value is outside
 // the bounds and above the upperbound.
 func (cb columnBounds) Between(ctx context.Context, nbf *types.NomsBinFormat, val types.Value) (ok bool, over bool, err error) {
+	// Only boundCase_isNull matches NULL values,
+	// otherwise we terminate the range scan.
+	// This is checked early to bypass unpredictable
+	// null type comparisons.
+	if val.Kind() == types.NullKind {
+		switch {
+		case cb.boundsCase == boundsCase_isNull:
```

Comment:
I'd rather this be an `if-else` since it's just two cases, and we're not using the switch as a type switch or anything, but it's nbd either way

Could even do
```
isNullCase := cb.boundsCase == boundsCase_isNull
return isNullCase, !isNullCase, nil
```
Which would remove that branch altogether and should be a tad faster